widgetastic_manageiq/__init__.py

```diff
@@ -4437,3 +4437,150 @@ def fill(self, value):
+
+    def expand_map_datastores(self, map_name):
+        """Click on datastores to expand to show details."""
+        el = self._get_map_element(map_name)
+        self.browser.click(self.browser.element(self.ITEM_DATASTORES_LOCATOR, parent=el))
+
+    def expand_map_associated_plans(self, map_name):
+        """Click on associated plans to expand to show details."""
+        try:
+            el = self._get_map_element(map_name)
+            self.browser.click(self.browser.element(
+                self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR, parent=el))
+        except NoSuchElementException:
+            raise ItemNotFound("There are no associated plans with map {}".format(map_name))
+
+    def read(self):
+        return self.all_items
+
+    def get_map_status(self, map_name):
+        el = self._get_map_element(map_name)
+        return self.browser.text(self.browser.element(self.ITEM_STATUS_LOCATOR,
+             parent=el))
+
+    def _get_sources(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.SOURCE_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def _get_targets(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.TARGET_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def get_map_source_clusters(self, map_name):
+        if not self.is_clusters_expanded(map_name):
+            self.expand_map_clusters(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_clusters(self, map_name):
+        if not self.is_clusters_expanded(map_name):
+            self.expand_map_clusters(map_name)
+        return self._get_targets(map_name)
+
+    def get_map_source_networks(self, map_name):
+        if not self.is_networks_expanded(map_name):
+            self.expand_map_networks(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_networks(self, map_name):
+        if not self.is_networks_expanded(map_name):
+            self.expand_map_networks(map_name)
+        return self._get_targets(map_name)
+
+    def get_map_source_datastores(self, map_name):
+        if not self.is_datastores_expanded(map_name):
+            self.expand_map_datastores(map_name)
+        return self._get_sources(map_name)
+
+    def get_map_target_datastores(self, map_name):
+        if not self.is_datastores_expanded(map_name):
+            self.expand_map_datastores(map_name)
+        return self._get_targets(map_name)
+
+    def get_associated_plans_count(self, map_name):
+        try:
+            el = self._get_map_element(map_name)
+            return self.browser.text(self.browser.element(self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR,
+                 parent=el))
+        except NoSuchElementException:
+            # Plan with no associated plans
+            return None
```

Comment:
Since the function is for `get_<>_count`, I think this should return 0 when there's no button so its a predictable return type.

Also, what's the need/use for this method, when there is `get_associated_plans`, and any caller could just look at the length of that return.

Reply:
This method will help me read the count which is displayed on the top bar in infra mappings list and then with get_associated_plans I can look at list length and validate that count is correctly displayed on the screen. Attaching a screenshot of how it looks on the page. 
![image](https://user-images.githubusercontent.com/8180469/41418847-a19a0790-6fbe-11e8-9ce4-89fa9c249256.png)

Reply:
Picture worth 1000 words :)